bdk_core/tests/test_sparse_chain.rs

```diff
@@ -291,6 +291,8 @@ fn can_introduce_older_checkpoints() {
 
 #[test]
 fn fix_blockhash_before_agreement_point() {
+    // TODO: This really shouldn't happen. Should we return error if this happens as this may
+    // require to total resync?
```

Comment:
I don't think we can return an error because we can't detect it happening? If we can consider making an issue instead of this comment.